src/pkg/services/m365/api/lists.go

```diff
@@ -529,6 +551,62 @@ func setAdditionalDataByColumnNames(
 	return filteredData
 }
 
+func hasAddressFields(additionalData map[string]any) (map[string]any, string, bool) {
+	for fieldName, value := range additionalData {
+		nestedFields, ok := value.(map[string]any)
+		if !ok || keys.HasKeys(nestedFields, GeoLocFieldName) {
+			continue
+		}
+
+		if keys.HasKeys(nestedFields, addressFieldNames...) {
+			return nestedFields, fieldName, true
+		}
+	}
+
+	return nil, "", false
+}
+
+func concatenateAddressFields(addressFields map[string]any) string {
+	concatenatedAddress := ""
+
+	if dispName, ok := addressFields[DisplayNameFieldName].(*string); ok {
+		concatenatedAddress = fmt.Sprintf("%s,%s", concatenatedAddress, ptr.Val(dispName))
+	}
+
+	if address, ok := addressFields[AddressFieldName].(map[string]any); ok {
+		concatenatedAddress = concatenateField(concatenatedAddress, address, NestedStreetFieldName)
+		concatenatedAddress = concatenateField(concatenatedAddress, address, NestedCityFieldName)
+		concatenatedAddress = concatenateField(concatenatedAddress, address, NestedStateFieldName)
+		concatenatedAddress = concatenateField(concatenatedAddress, address, NestedCountryFieldName)
+		concatenatedAddress = concatenateField(concatenatedAddress, address, NestedPostalCodeFieldName)
+	}
+
+	if coords, ok := addressFields[CoordinatesFieldName].(map[string]any); ok {
+		concatenatedAddress = concatenateField(concatenatedAddress, coords, NestedLatitudeFieldName)
+		concatenatedAddress = concatenateField(concatenatedAddress, coords, NestedLongitudeFieldName)
+	}
+
+	if len(concatenatedAddress) > 0 {
+		return concatenatedAddress[1:]
+	}
+
+	return ""
+}
+
+func concatenateField(concatenatedAddress string, field map[string]any, fieldName string) string {
+	if _, ok := field[fieldName]; !ok {
+		return concatenatedAddress
+	}
+
+	if ptrValue, ok := field[fieldName].(*string); ok {
+		concatenatedAddress = fmt.Sprintf("%s,%s", concatenatedAddress, ptr.Val(ptrValue))
+	} else if ptrValue, ok := field[fieldName].(*float64); ok {
+		concatenatedAddress = fmt.Sprintf("%s,%v", concatenatedAddress, ptr.Val(ptrValue))
+	}
+
+	return concatenatedAddress
```

Comment:
three suggestions:
1/ this should use the str package for the first check.
2/ the not-exists check is redundant.
3/ the more idiomatic way to concatenate strings is to build a slice and then Join it.

```suggestion
func concatenateField(
parts []string,
fields map[string]any,
key string,
 ) []string {
	if v, err := str.AnyValueToString(key, fields); err != nil {
		parts = append(parts, v)
	} else if fl, ok := field[fieldName].(*float64); ok {
		parts = append(parts, fmt.Sprintf("%v", fl))
	}

	return parts
	// the caller can later use strings.Join(parts, ",")
```